.github/workflows/py-tests.yml

```diff
@@ -33,7 +33,6 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: '3.10'
-        if: ${{ runner.os == 'Windows' }}
```

Comment:
Christian might know this already, but just in case: This line (deleted in this PR) was added by me specifically to make the macOS tests pass. The GH Actions runners have Python pre-installed, so the existence of the extra Python seemed to confuse CMake. The test failures we are seeing know are probably the same as what blocked me for more than a month.

Reply:
You are probably right here. The strange thing is that cmake does not even use the version installed by the github action (3.10), but uses some other version that it finds (3.8). I don't know how to resolve this... The default Python version in the MacOS runner is now 3.11 which our Python target is not compatible to. I will continue trial and error and see where it gets me...

Peter, you know the Python target better than me: What do you think how much effort it would be to actually fix the errors that we see popping up in CI when Python 3.11 is used?

Reply:
I just noticed that I made a mistake and excluded version 3.10, which explains why cmake selected version 3.8 (which seems to be preinstalled as well)